From a0d50f3497facfaff2ee30ea46d3709371d1c4cd Mon Sep 17 00:00:00 2001
From: Andreas Monzner <andreas.monzner@dream-property.net>
Date: Wed, 27 Feb 2013 22:26:37 +0100
Subject: [PATCH] NFS: limit max rwsize to 8k from client side

STB's nic's generally do not implement scatter/gather
support. This, combined with the default huge rwsize
mount option of NFS, leads to memory fragementation
issues (i.e. OOM killer, crashes). Work around this
issue by limiting the rwsize to 8k, which is safe on
all stb's currently supported. Issue resolved with
help from pieterg and sjaaky, thanks.
---
 include/linux/nfs_xdr.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/nfs_xdr.h b/include/linux/nfs_xdr.h
index 41116ab..181ef86 100644
--- a/include/linux/nfs_xdr.h
+++ b/include/linux/nfs_xdr.h
@@ -11,7 +11,7 @@
  * support a megabyte or more.  The default is left at 4096 bytes, which is
  * reasonable for NFS over UDP.
  */
-#define NFS_MAX_FILE_IO_SIZE	(1048576U)
+#define NFS_MAX_FILE_IO_SIZE	(8192)
 #define NFS_DEF_FILE_IO_SIZE	(4096U)
 #define NFS_MIN_FILE_IO_SIZE	(1024U)
 
-- 
1.7.10.4

